### PR TITLE
[USER]: implement user repository, service and controller layer.

### PR DESCRIPTION
**Pull request by IhorKalaur**

### CODE REVIEW

File: src/main/java/com/team01/carsharingapp/dto/user/UpdateUserDto.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+public class UpdateUserDto {
+    @NotBlank(message = "required field")
+    @Size(min = 5, max = 50, message = "size must be between 5 and 50 character")
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/dto/user/UpdateUserDto.java:11]: Set min to 4

File: src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -40,7 +46,54 @@ public User getCurrentUser() {
...
+    @Transactional
+    public UserDto update(UpdateUserDto updateUserDto) {
+        User user = userMapper.toModel(updateUserDto);
+
+        user.setId(getCurrentUser().getId());
+        user.setPassword(passwordEncoder.encode(updateUserDto.getPassword()));
+
+        return convertToDto(userRepository.save(user));
+    }
+
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java:76]: Update current user, not create new with same id

File: src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -40,7 +46,54 @@ public User getCurrentUser() {
...
+    public UserDto getInfo() {
+        return convertToDto(getUserById(getCurrentUser().getId()));
+    }
+
+    @Override
+    @Transactional
+    public UserDto updateRoles(Long id, UpdateUserRoleDto updateRoleDto) {
+        User user = getUserById(id);
+        user.setRoles(updateRoleDto.roleIds().stream()
+                .map(this::getRoleById)
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java:60]: You can add only id, not get entire role from DB

File: src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -40,7 +46,54 @@ public User getCurrentUser() {
...
+                        + id));
+    }
+
+    private UserDto convertToDto(User user) {
+        UserDto dto = userMapper.toDto(user);
+        dto.setRoleIds(user.getRoles().stream()
+                .map(Role::getId)
+                .collect(Collectors.toSet()));
+        return dto;
+    }
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java:98]: You can do it in mapper

**Review by Someboty — COMMENTED** [review]
GJ! Left few comments

**Review by Someboty — APPROVED** [review]

**Review by keda-vlad — APPROVED** [review]

**Pull request merged**